scalalib/src/ScalaModule.scala

```diff
@@ -185,10 +185,15 @@ trait ScalaModule extends JavaModule { outer =>
       }
     }
 
-    val files = allSourceFiles().map(_.path.toString)
+    val files = 
+      if (isDottyOrScala3(scalaVersion()) && useScaladocInScala3(scalaVersion()))
+        // scaladoc 3.x required list of tasty files, jars or classes directories
+        compile().classes.path.toString :: Nil
+      else 
+        allSourceFiles().map(_.path.toString)
 
     val outputOptions =
-      if (isDottyOrScala3(scalaVersion()))
+      if (isDottyOrScala3(scalaVersion()) && !useScaladocInScala3(scalaVersion()))
         Seq("-siteroot", javadocDir.toNIO.toString)
       else
         Seq("-d", javadocDir.toNIO.toString)
```

Comment:
I think we need a separate condition here for scaladoc3, which adds a separate `-d` and `-siteroot` parameter. (at least when I tested scaladoc 3 I needed to pass those flags in)